src/components/Invoice/use-invoice.ts

```diff
@@ -0,0 +1,20 @@
+import { useQuery } from '@tanstack/react-query';
+
+const getInvoice =
+  (id: string) =>
+  ({ signal }: { signal: AbortSignal }) =>
+    fetch(`/api/invoices/${id}`, { signal })
+      .then((res) => {
+        if (res.status > 299) {
```

Comment:
It doesn't matter in that case, but I would rather set > 399 here, just for readability.